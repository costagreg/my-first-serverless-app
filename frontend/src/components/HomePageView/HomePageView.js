import React from 'react'
import { connect } from 'react-redux'
import UrlInputForm from '../../components/UrlInputForm'
import UrlShortenerOutput from '../../components/UrlShortenerOutput'
import LoginForm from '../../components/LoginForm'
import { shortUrl } from '../../actions/shortener'

import './homePageView.scss'

const mapToProps = (state) => ({
  error: state.shortener.error,
  urlShortened: state.shortener.urlShortened,
})

const HomePageView = ({ shortUrl, urlShortened, error }) => (
  <div className="homePageView homePageView--initialState">
    <center>
      <h2>URL Shortener!</h2>
      <p>Simply copy & paste your link and click the cut button!</p>
    </center>
    <UrlInputForm shortUrl={shortUrl} error={error} />
    <UrlShortenerOutput error={error} urlShortened={urlShortened} />
    <LoginForm />
  </div>
)

HomePageView.propTypes = {}

export default connect(mapToProps, { shortUrl })(HomePageView)
