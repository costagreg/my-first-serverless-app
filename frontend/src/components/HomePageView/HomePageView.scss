.homePageView {
  &--initialState{
    padding-top: 7%;
  }
}