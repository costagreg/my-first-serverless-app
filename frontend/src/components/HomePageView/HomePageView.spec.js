import React from 'react'
import HomePageView from './HomePageView'
import { render } from '@testing-library/react'

describe('HomePageView', () => {
  describe('@renders', () => {
    it('renders <HomePageView /> without any error', () => {
      const { asFragment } = render(<HomePageView />)

      expect(asFragment()).toMatchSnapshot()
    })
  })
})
