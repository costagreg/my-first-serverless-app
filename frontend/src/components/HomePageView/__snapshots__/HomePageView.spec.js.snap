// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePageView @renders renders <HomePageView /> without any error 1`] = `
<DocumentFragment>
  <div
    class="homePageView homePageView--initialState"
  >
    <center>
      <h2>
        URL Shortener!
      </h2>
      <p>
        Simply copy & paste your link and click the cut button!
      </p>
    </center>
    <form
      class="urlInputForm"
    >
      <input
        class="urlInputForm__input"
        placeholder="https://www.example.com"
        type="text"
        value=""
      />
      <button
        class="urlInputForm__submit"
        type="submit"
      >
        Cut me!
      </button>
    </form>
    <div
      class="urlShortenerOutput"
    >
      <textarea
        class="urlShortenerOutput__link"
        readonly=""
      />
    </div>
    <div
      class="loginForm"
    >
      <form
        class="loginForm__formContainer"
      >
        <p>
          Signup to track your link!
        </p>
        <input
          class="loginForm__email"
          placeholder="Email"
        />
        <input
          class="loginForm__password"
          placeholder="Password"
        />
        <div
          class="loginForm__buttons"
        >
          <button
            class="loginForm__loginButton"
          >
            Login
          </button>
          <button
            class="loginForm__signupButton"
          >
            Signup
          </button>
        </div>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
